### DISCUSSION TITLE
Documentation: php quickstart tool

### DISCUSSION THREAD

**Discussion by GruppoCO** [spatie/laravel-google-calendar#265] [Q&A]
Hi,
I need to authenticate with OAuth2, but it seems that the [php quickstart tool](https://developers.google.com/calendar/quickstart/php) link doesn't exists anymore. How can I get the credentials.json and token.json?

**Comment by NowakAdmin** [comment]
credentials json -> https://console.cloud.google.com/apis/api/calendar-json.googleapis.com/credentials

i cant get token.json, php quickstart tool vanished, there are only javascript page to test aplication - I can log on mine and get events list in google demo but I can't in laravel.

**Reply by egvoncannon** [reply]
You can get the token.json from [https://developers.google.com/oauthplayground](url). Choose "Use your own credentials" in the settings, provide them, and complete steps 1 and 2 this will show the JSON for the token like:
{
    "access_token": "",
    "scope": "https://www.googleapis.com/auth/calendar.settings.readonly ",
    "token_type": "Bearer",
    "expires_in": 3599,
    "refresh_token": ""
}

Copy this and put it in the oauth-token.json file. You do not need to complete step 3.

**Reply by Azeempro2** [reply]
Hi @egvoncannon where I can get oauth-credentials.json

**Reply by AIC-BV** [reply]
Hi @egvoncannon, doesn't this expire after one hour?
I copy pasted the reply but after a while the playground says that it is expired.

**Reply by nxmndr** [reply]
This doesn't seem to work anymore. After the selecting scopes step, you pick an account, then are redirected to "an error happened" page and then to your account homepage without further explanation.

**Reply by nxmndr** [reply]
Why is this such a PAIN in freaking 2024.

**Comment by theafolayan** [comment]
Hi @sebastiandedeyne  and @freekmurze  It appears the credentials setup guide is out of date as I'm facing this issue as well.
I'd like to create a PR updating the setup part of the ReadMe.md so it matches what new users of the package should expect to see. What do you think?

**Reply by freekmurze** [reply]
Feel free to send a PR.

**Reply by nxmndr** [reply]
@freekmurze If you are not able to maintain compatibility (understandable, life and all that), which is the point of the package, then please mark it as unmaintained. People loose hours (and hair) on this.

**Comment by hmrst** [comment]
@theafolayan do you have a solution in how to get the token json file or the quickstart guide?

**Comment by esghendry** [comment]
Still no fix... Anyone?

**Comment by theafolayan** [comment]
We have a fix and would be making it open source soon.



Oluwaseun Raphael Oluwaseun

**Reply by sukhrobnurali** [reply]
is it ready? one year passed, where it is?

**Comment by esghendry** [comment]
> We have a fix and would be making it open source soon. Oluwaseun Raphael Oluwaseun
> […](#)
> On Thu, May 18, 2023, 12:21 PM hn3 ***@***.***> wrote: Still no fix... Anyone? — Reply to this email directly, view it on GitHub <[#265 (comment)](https://github.com/spatie/laravel-google-calendar/discussions/265#discussioncomment-5938001)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKRIOXUM5BETP6YBEH566YTXGYA45ANCNFSM6AAAAAAU5BPWEM> . You are receiving this because you were mentioned.Message ID: <spatie/laravel-google-calendar/repo-discussions/265/comments/5938001@ github.com>

Great, but what does your 'soon' mean?

**Comment by AliN11** [comment]
I cannot start using this package because of this error. Any ideas?

**Reply by egvoncannon** [reply]
See my response and NowakAdmin's above to get token and credentials from Google.

**Comment by egvoncannon** [comment]
If you don't use your own credentials it will assume you are testing and
expire after one hour. Ideally you should create a service account and
provide it's credentials.

On Mon, Sep 25, 2023, 8:06 AM AIC BV ***@***.***> wrote:

> Hi @egvoncannon <https://github.com/egvoncannon>, doesn't this expire
> after one hour?
> I copy pasted the reply but after a while the playground says that it is
> expired.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spatie/laravel-google-calendar/discussions/265#discussioncomment-7100945>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A27CUNJX74BYMQYDF3LZ6OTX4FXSPANCNFSM6AAAAAAU5BPWEM>
> .
> You are receiving this because you were mentioned.Message ID:
> <spatie/laravel-google-calendar/repo-discussions/265/comments/7100945@
> github.com>
>

-- 


All email correspondence to and from this address is subject to the North 
Carolina Public Records Law, which may result in monitoring and disclosure 
to third parties, including law enforcement.

**Reply by EmilioPuljiz** [reply]
Hello, I'm asking you, I was doing a little research, the technique that you put at the beginning of the comments worked for me, the problem is that I cannot refresh the token, normally the token expires after 60 minutes, after that the package does not have a function to refresh, I tried to do it manually, but it keeps throwing me an error.

Google\Service\Exception
{ "error": "invalid_grant", "error_description": "Token has been expired or revoked." }

I clarify that everything works fine for me while I use the token that I generated with the official Google documentation, I log in, I get the token, I place it in storage/app/google-calendar/oauth-token.json
All this works correctly, what happens to me is that the token expires and I cannot consult or create more events in the calendar

**Comment by egvoncannon** [comment]
Honestly, it has been a while since I worked with the Oauth playground. As
I recall, providing your own credentials is how you stop the token from
expiring after an hour. I am sure others have more, and more recent,
experience with the issue. I posted here in hopes it would help others with
the same issue. Please provide feedback if you solve your issue. Good luck!

On Wed, Nov 29, 2023 at 4:24 PM Emilio Puljiz ***@***.***>
wrote:

> Hello, I'm asking you, I was doing a little research, the technique that
> you put at the beginning of the comments worked for me, the problem is that
> I cannot refresh the token, normally the token expires after 60 minutes,
> after that the package does not have a function to refresh, I tried to do
> it manually, but it keeps throwing me an error.
>
> Google\Service\Exception
> { "error": "invalid_grant", "error_description": "Token has been expired
> or revoked." }
>
> I clarify that everything works fine for me while I use the token that I
> generated with the official Google documentation, I log in, I get the
> token, I place it in storage/app/google-calendar/oauth-token.json
> All this works correctly, what happens to me is that the token expires and
> I cannot consult or create more events in the calendar
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spatie/laravel-google-calendar/discussions/265#discussioncomment-7710344>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A27CUNOZMHRSXBFIE5XV7ALYG6RXFAVCNFSM6AAAAAAU5BPWEOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOMJQGM2DI>
> .
> You are receiving this because you were mentioned.Message ID:
> <spatie/laravel-google-calendar/repo-discussions/265/comments/7710344@
> github.com>
>


-- 
 Eric VonCannon
Technician/Jr. Programmer
Technology Dept.
***@***.***
828-890-1658 ext.16026

-- 


All email correspondence to and from this address is subject to the North 
Carolina Public Records Law, which may result in monitoring and disclosure 
to third parties, including law enforcement.

**Comment by cornelpopa** [comment]
I have the same issue... I had to switch from service account to oauth2 and I have realized that php quickstart is no longer available.
I have no idea how to generate the credentials.json and token.json. Any tocken expires in 60 minutes...  Has anybody found a solution?

**Reply by EmilioPuljiz** [reply]
Hello, I had several problems similar to yours, I switched to using the plugin https://github.com/googleapis/google-api-php-client

I'm doing great, I was able to create everything! If you need help write to me on my linkedIn https://www.linkedin.com/in/emilio-puljiz/

**Comment by ana-lisboa** [comment]
I've only tried this in my local machine. But It looks like you can either go with the service account json credentials OR with the oauth2.

I say this because I was stuck with the broken oauth2 links, but had setup my service account json. Decided to just give it a go, and can at least list all events from the google calendar!
